var/ramble/repos/builtin/workflow_managers/slurm/workflow_manager.py

```diff
@@ -0,0 +1,197 @@
+        return rf"""
+#!/bin/bash
+{_ensure_job_id_snippet}
+{self.runner.generate_cancel_command("$job_id")}
+        """.strip()
+
+    render_content(
+        name="slurm_execute_experiment", content_func="_slurm_execute_script"
+    )
+
+    def _slurm_execute_script(self):
+        headers = ["#!/bin/bash"]
+        expander = self.app_inst.expander
+        # Adding pre-defined sets of headers
+        pragmas = [
+            ("#SBATCH -N {}", "n_nodes"),
+            ("#SBATCH -p {}", "partition"),
+            ("#SBATCH --ntasks-per-node {}", "processes_per_node"),
+            ("#SBATCH -J {}", "job_name"),
+        ]
+        for tpl, var in pragmas:
+            try:
+                val = expander.expand_var_name(var, allow_passthrough=False)
+            except ExpanderError:
+                # Skip a particular header if any of the vars are not defined
+                continue
+            if val:
+                headers.append(tpl.format(val))
+        # Adding extra arbitrary headers
+        try:
+            extra_sbatch_headers_raw = expander.expand_var_name(
+                "extra_sbatch_headers", allow_passthrough=False
+            )
+            extra_sbatch_headers = extra_sbatch_headers_raw.strip().split("\n")
+            extra_headers = [
+                expander.expand_var(h) for h in extra_sbatch_headers
+            ]
+            headers = headers + extra_headers
+        except ExpanderError:
+            pass
+        header_str = "\n".join(headers)
+        content = rf"""
+{header_str}
+
+cd {{experiment_run_dir}}
+
+{self.runner.generate_hostfile_command()} > {{experiment_run_dir}}/hostfile
+{{command}}
+        """.strip()
+
+        return content
+
+    def get_status(self, workspace):
+        expander = self.app_inst.expander
+        run_dir = expander.expand_var_name("experiment_run_dir")
+        job_id_file = os.path.join(run_dir, ".slurm_job")
+        status = experiment_status.UNKNOWN
+        if not os.path.isfile(job_id_file):
+            logger.warn("job_id file is missing")
+            return status
+        self.runner.set_dry_run(workspace.dry_run)
+        self.runner.set_run_dir(run_dir)
+        wm_status = self.runner.get_status()
+        if wm_status is not None and hasattr(experiment_status, wm_status):
+            status = getattr(experiment_status, wm_status)
+        return status
+
+
+class SlurmRunner:
+    """Runner for executing slurm commands"""
+
+    def __init__(self, dry_run=False):
+        self.dry_run = dry_run
+        self.run_dir = None
+
+    def set_dry_run(self, dry_run=False):
+        """
+        Set the dry_run state of this runner
+        """
+        self.dry_run = dry_run
+
+    def set_run_dir(self, run_dir):
+        """
+        Set the experiment_run_dir of this runner
+        """
+        self.run_dir = run_dir
+
+    def generate_query_command(self, job_id):
+        return rf"""
+status=$(squeue -h -o "%t" -j {job_id})
```

Comment:
Good call. I moved the get_status call inside the runner to be in python instead of relying on this bash script, so now `ramble workspace analyze` would rely on python. Do you prefer even the `--executor "{batch_query}"` to be handled in python instead of the bash script right now?